.github/workflows/publish-to-npm.yml

```diff
@@ -0,0 +1,95 @@
+name: publish-to-npm
+on:
+  release:
+    types: [released]
+  # Support manual releases in case something goes wrong, or we need to do a test.
+  workflow_dispatch:
+    inputs:
+      tag:
+        description: Tag to be published
+        type: string
+        required: true
+
+jobs:
+  # Step 1: Verify that the tag we're trying to release is a valid candidate for publishing.
+  verify-candidate-tag:
+    runs-on: ubuntu-latest
+    steps:
+      # Check out the release branch, and get its head commit as output for later.
+      - uses: actions/checkout@v3
+        with:
+          ref: 'release-3'
+      - run: echo "COMMIT_ID=$(git rev-parse HEAD)" >> $GITHUB_OUTPUT
+        id: get-branch-commit
+      # Checkout the tag we want to release, and get its head commit as output for later.
+      - uses: actions/checkout@v3
+        with:
+          ref: ${{ github.event.release.tag_name || inputs.tag }}
+      - run: echo "COMMIT_ID=$(git rev-parse HEAD)" >> $GITHUB_OUTPUT
+        id: get-tag-commit
+      # If the two commits aren't identical, the tag isn't eligible for release.
+      - name: Fail non-matching commits
+        if: ${{ steps.get-branch-commit.outputs.COMMIT_ID != steps.get-tag-commit.outputs.COMMIT_ID }}
+        run: |
+          echo "Tag commit must match latest commit in release-3. Branch is ${{ steps.get-branch-commit.outputs.COMMIT_ID }}. Tag is ${{ steps.get-tag-commit.outputs.COMMIT_ID }}"
+          exit 1
+      # Verify that the tag's value matches the version in package.json.
+      - name: Compare tag to package.json
+        run: |
+          TAG=${{ github.event.release.tag_name || inputs.tag }}
+          PACKAGE_VERSION=v`cat package.json | jq '.version' | xargs`
+          [[ ${TAG} == ${PACKAGE_VERSION} ]] || (echo "Tag name must match package.json version, prefixed by lowercase v" && exit 1)
+  # Step 2: Publish the tag as a release candidate.
+  publish-rc:
```

Comment:
This job is analogous to an example job provided by the CLI team, with the only significant difference being that we use the `latest-rc` tag, whereas they use `latest`.